// @strapi/admin/build/2077.5456ccd1.chunk.js
(self.webpackChunk_strapi_admin=self.webpackChunk_strapi_admin||[]).push([[2077],{74201:(v,j,g)=>{"use strict";v.exports=g(45214)},45214:function(v,j,g){(function(m,b){v.exports=b(g(32735),g(19615))})(this,function(m,b){return function(t){var f={};function e(i){if(f[i])return f[i].exports;var r=f[i]={i,l:!1,exports:{}};return t[i].call(r.exports,r,r.exports,e),r.l=!0,r.exports}return e.m=t,e.c=f,e.d=function(i,r,c){e.o(i,r)||Object.defineProperty(i,r,{enumerable:!0,get:c})},e.r=function(i){typeof Symbol!="undefined"&&Symbol.toStringTag&&Object.defineProperty(i,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(i,"__esModule",{value:!0})},e.t=function(i,r){if(1&r&&(i=e(i)),8&r||4&r&&typeof i=="object"&&i&&i.__esModule)return i;var c=Object.create(null);if(e.r(c),Object.defineProperty(c,"default",{enumerable:!0,value:i}),2&r&&typeof i!="string")for(var a in i)e.d(c,a,function(d){return i[d]}.bind(null,a));return c},e.n=function(i){var r=i&&i.__esModule?function(){return i.default}:function(){return i};return e.d(r,"a",r),r},e.o=function(i,r){return Object.prototype.hasOwnProperty.call(i,r)},e.p="",e(e.s=64)}([function(t,f,e){t.exports=e(19)()},function(t,f){t.exports=m},function(t,f){t.exports=b},function(t,f){t.exports=function(e,i){return i||(i=e.slice(0)),Object.freeze(Object.defineProperties(e,{raw:{value:Object.freeze(i)}}))},t.exports.default=t.exports,t.exports.__esModule=!0},function(t,f,e){var i=e(24);t.exports=function(r,c){if(r==null)return{};var a,d,p=i(r,c);if(Object.getOwnPropertySymbols){var h=Object.getOwnPropertySymbols(r);for(d=0;d<h.length;d++)a=h[d],c.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(r,a)&&(p[a]=r[a])}return p},t.exports.default=t.exports,t.exports.__esModule=!0},function(t,f){function e(){return t.exports=e=Object.assign||function(i){for(var r=1;r<arguments.length;r++){var c=arguments[r];for(var a in c)Object.prototype.hasOwnProperty.call(c,a)&&(i[a]=c[a])}return i},t.exports.default=t.exports,t.exports.__esModule=!0,e.apply(this,arguments)}t.exports=e,t.exports.default=t.exports,t.exports.__esModule=!0},function(t,f,e){"use strict";e.r(f),e.d(f,"Box",function(){return S});var i,r=e(3),c=e.n(r),a=e(2),d=e.n(a),p=e(7),h=e(1),y=e.n(h),w=e(0),u=e.n(w),l=function(n){return y.a.createElement("div",n)},x={background:void 0,borderColor:void 0,color:void 0,hiddenS:!1,hiddenXS:!1,padding:void 0,paddingTop:void 0,paddingRight:void 0,paddingBottom:void 0,paddingLeft:void 0,hasRadius:!1,shadow:void 0,children:null,shrink:void 0,grow:void 0,basis:void 0,flex:void 0,_hover:function(){}},O={_hover:u.a.func,background:u.a.string,basis:u.a.oneOfType([u.a.string,u.a.string]),borderColor:u.a.string,children:u.a.oneOfType([u.a.node,u.a.string]),color:u.a.string,flex:u.a.oneOfType([u.a.string,u.a.string]),grow:u.a.oneOfType([u.a.string,u.a.string]),hasRadius:u.a.bool,hiddenS:u.a.bool,hiddenXS:u.a.bool,padding:u.a.oneOfType([u.a.number,u.a.arrayOf(u.a.number)]),paddingBottom:u.a.oneOfType([u.a.number,u.a.arrayOf(u.a.number)]),paddingLeft:u.a.oneOfType([u.a.number,u.a.arrayOf(u.a.number)]),paddingRight:u.a.oneOfType([u.a.number,u.a.arrayOf(u.a.number)]),paddingTop:u.a.oneOfType([u.a.number,u.a.arrayOf(u.a.number)]),shadow:u.a.string,shrink:u.a.oneOfType([u.a.string,u.a.string])};l.defaultProps=x,l.propTypes=O;var T={color:!0},S=d.a.div.withConfig({shouldForwardProp:function(n,o){return!T[n]&&o(n)}})(i||(i=c()([`
  // Font
  font-size: `,`;

  // Colors
  background: `,`;
  color: `,`;

  // Spaces
  `,`
  `,`
  `,`
  `,`
  `,`
  `,`
  `,`
  `,`
  `,`

  // Responsive hiding
  `,`
  `,`
  

  // Borders
  border-radius: `,`;
  border-style: `,`;
  border-width: `,`;
  border-color: `,`;
  border: `,`;

  // Shadows
  box-shadow: `,`;

  // Handlers
  pointer-events: `,`;
  &:hover {
    `,`
  }

  // Display
  display: `,`;

  // Position
  position: `,`;
  left: `,`;
  right: `,`;
  top: `,`;
  bottom: `,`;
  z-index: `,`;
  overflow: `,`;
  cursor: `,`;

  // Size
  width: `,`;
  max-width: `,`;
  min-width: `,`;
  height: `,`;
  max-height: `,`;
  min-height: `,`;

  // Animation
  transition: `,`;
  transform: `,`;
  animation: `,`;

  //Flexbox children props
  flex-shrink: `,`;
  flex-grow: `,`;
  flex-basis: `,`;
  flex: `,`;

  // Text
  text-align: `,`;
  text-transform: `,`;
  line-height: `,`;

  // Cursor
  cursor: `,`;
`])),function(n){var o=n.fontSize;return n.theme.fontSizes[o]||o},function(n){var o=n.theme,s=n.background;return o.colors[s]},function(n){var o=n.theme,s=n.color;return o.colors[s]},function(n){var o=n.theme,s=n.padding;return Object(p.a)("padding",s,o)},function(n){var o=n.theme,s=n.paddingTop;return Object(p.a)("padding-top",s,o)},function(n){var o=n.theme,s=n.paddingRight;return Object(p.a)("padding-right",s,o)},function(n){var o=n.theme,s=n.paddingBottom;return Object(p.a)("padding-bottom",s,o)},function(n){var o=n.theme,s=n.paddingLeft;return Object(p.a)("padding-left",s,o)},function(n){var o=n.theme,s=n.marginLeft;return Object(p.a)("margin-left",s,o)},function(n){var o=n.theme,s=n.marginRight;return Object(p.a)("margin-right",s,o)},function(n){var o=n.theme,s=n.marginTop;return Object(p.a)("margin-top",s,o)},function(n){var o=n.theme,s=n.marginBottom;return Object(p.a)("margin-bottom",s,o)},function(n){var o=n.theme;return n.hiddenS?"".concat(o.mediaQueries.tablet," { display: none; }"):void 0},function(n){var o=n.theme;return n.hiddenXS?"".concat(o.mediaQueries.mobile," { display: none; }"):void 0},function(n){var o=n.theme,s=n.hasRadius,C=n.borderRadius;return s?o.borderRadius:C},function(n){return n.borderStyle},function(n){return n.borderWidth},function(n){var o=n.borderColor;return n.theme.colors[o]},function(n){var o=n.theme,s=n.borderColor,C=n.borderStyle,P=n.borderWidth;if(s&&!C&&!P)return"1px solid ".concat(o.colors[s])},function(n){var o=n.theme,s=n.shadow;return o.shadows[s]},function(n){return n.pointerEvents},function(n){var o=n._hover,s=n.theme;return o?o(s):void 0},function(n){return n.display},function(n){return n.position},function(n){var o=n.left;return n.theme.spaces[o]||o},function(n){var o=n.right;return n.theme.spaces[o]||o},function(n){var o=n.top;return n.theme.spaces[o]||o},function(n){var o=n.bottom;return n.theme.spaces[o]||o},function(n){return n.zIndex},function(n){return n.overflow},function(n){return n.cursor},function(n){var o=n.width;return n.theme.spaces[o]||o},function(n){var o=n.maxWidth;return n.theme.spaces[o]||o},function(n){var o=n.minWidth;return n.theme.spaces[o]||o},function(n){var o=n.height;return n.theme.spaces[o]||o},function(n){var o=n.maxHeight;return n.theme.spaces[o]||o},function(n){var o=n.minHeight;return n.theme.spaces[o]||o},function(n){return n.transition},function(n){return n.transform},function(n){return n.animation},function(n){return n.shrink},function(n){return n.grow},function(n){return n.basis},function(n){return n.flex},function(n){return n.textAlign},function(n){return n.textTransform},function(n){return n.lineHeight},function(n){return n.cursor});S.defaultProps=x,S.propTypes=O},function(t,f,e){"use strict";var i=e(10),r=e.n(i),c=e(13),a=e.n(c);f.a=function(d,p,h){var y=p;if(Array.isArray(p)||a()(p)!=="object"||(y=[p==null?void 0:p.desktop,p==null?void 0:p.tablet,p==null?void 0:p.mobile]),y!==void 0){if(Array.isArray(y)){var w=y,u=r()(w,3),l=u[0],x=u[1],O=u[2],T="".concat(d,": ").concat(h.spaces[l],";");return x!==void 0&&(T+="".concat(h.mediaQueries.tablet,`{
          `).concat(d,": ").concat(h.spaces[x],`;
        }`)),O!==void 0&&(T+="".concat(h.mediaQueries.mobile,`{
          `).concat(d,": ").concat(h.spaces[O],`;
        }`)),T}var S=h.spaces[y]||y;return"".concat(d,": ").concat(S,";")}}},function(t,f,e){"use strict";e.r(f),e.d(f,"Typography",function(){return S});var i,r=e(3),c=e.n(r),a=e(2),d=e.n(a),p=["alpha","beta","delta","epsilon","omega","pi","sigma"],h=e(1),y=e.n(h),w=e(0),u=e.n(w),l=function(n){return y.a.createElement("div",n)},x={ellipsis:!1,fontWeight:void 0,fontSize:void 0,lineHeight:void 0,textColor:void 0,textTransform:void 0,variant:"omega"},O={ellipsis:u.a.bool,fontSize:u.a.oneOfType([u.a.number,u.a.string]),fontWeight:u.a.string,lineHeight:u.a.oneOfType([u.a.number,u.a.string]),textColor:u.a.string,textTransform:u.a.string,variant:u.a.oneOf(p)};l.defaultProps=x,l.propTypes=O;var T={fontSize:!0,fontWeight:!0},S=d.a.span.withConfig({shouldForwardProp:function(n,o){return!T[n]&&o(n)}})(i||(i=c()([`
  font-weight: `,`;
  font-size: `,`;
  line-height: `,`;
  color: `,`;
  text-transform: `,`;
  `,`
  `,`
`])),function(n){var o=n.theme,s=n.fontWeight;return o.fontWeights[s]},function(n){var o=n.theme,s=n.fontSize;return o.fontSizes[s]},function(n){var o=n.theme,s=n.lineHeight;return o.lineHeights[s]},function(n){var o=n.theme,s=n.textColor;return o.colors[s||"neutral800"]},function(n){return n.textTransform},function(n){return n.ellipsis&&`
    display: block;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  `},function(n){var o=n.variant,s=n.theme;switch(o){case"alpha":return`
        font-weight: `.concat(s.fontWeights.bold,`;
        font-size: `).concat(s.fontSizes[5],`;
        line-height: `).concat(s.lineHeights[2],`;
      `);case"beta":return`
        font-weight: `.concat(s.fontWeights.bold,`;
        font-size: `).concat(s.fontSizes[4],`;
        line-height: `).concat(s.lineHeights[1],`;
      `);case"delta":return`
        font-weight: `.concat(s.fontWeights.semiBold,`;
        font-size: `).concat(s.fontSizes[3],`;
        line-height: `).concat(s.lineHeights[2],`;
      `);case"epsilon":return`
        font-size: `.concat(s.fontSizes[3],`;
        line-height: `).concat(s.lineHeights[6],`;
      `);case"omega":return`
        font-size: `.concat(s.fontSizes[2],`;
        line-height: `).concat(s.lineHeights[4],`;
      `);case"pi":return`
        font-size: `.concat(s.fontSizes[1],`;
        line-height: `).concat(s.lineHeights[3],`;
      `);case"sigma":return`
        font-weight: `.concat(s.fontWeights.bold,`;
        font-size: `).concat(s.fontSizes[0],`;
        line-height: `).concat(s.lineHeights[5],`;
        text-transform: uppercase;
      `);default:return`
        font-size: `.concat(s.fontSizes[2],`;
      `)}});S.defaultProps=x,S.propTypes=O},function(t,f,e){"use strict";e.r(f),e.d(f,"Flex",function(){return n});var i,r=e(3),c=e.n(r),a=e(2),d=e.n(a),p=e(6),h=e(7),y=e(1),w=e.n(y),u=e(0),l=e.n(u),x=function(o){return w.a.createElement("div",o)},O={alignItems:"center",basis:void 0,direction:"row",gap:void 0,inline:!1,justifyContent:void 0,reverse:!1,wrap:void 0},T={alignItems:l.a.string,basis:l.a.oneOfType([l.a.string,l.a.number]),direction:l.a.string,gap:l.a.oneOfType([l.a.shape({desktop:l.a.number,mobile:l.a.number,tablet:l.a.number}),l.a.number,l.a.arrayOf(l.a.number),l.a.string]),inline:l.a.bool,justifyContent:l.a.string,reverse:l.a.bool,wrap:l.a.string};x.defaultProps=O,x.propTypes=T;var S={direction:!0},n=d()(p.Box).withConfig({shouldForwardProp:function(o,s){return!S[o]&&s(o)}})(i||(i=c()([`
  align-items: `,`;
  display: `,`;
  flex-direction: `,`;
  flex-wrap: `,`;
  `,`};
  justify-content: `,`;
`])),function(o){return o.alignItems},function(o){return o.inline?"inline-flex":"flex"},function(o){return o.direction},function(o){return o.wrap},function(o){var s=o.gap,C=o.theme;return Object(h.a)("gap",s,C)},function(o){return o.justifyContent});n.defaultProps=O,n.propTypes=T},function(t,f,e){var i=e(25),r=e(26),c=e(22),a=e(27);t.exports=function(d,p){return i(d)||r(d,p)||c(d,p)||a()},t.exports.default=t.exports,t.exports.__esModule=!0},,,function(t,f){function e(i){return typeof Symbol=="function"&&typeof Symbol.iterator=="symbol"?(t.exports=e=function(r){return typeof r},t.exports.default=t.exports,t.exports.__esModule=!0):(t.exports=e=function(r){return r&&typeof Symbol=="function"&&r.constructor===Symbol&&r!==Symbol.prototype?"symbol":typeof r},t.exports.default=t.exports,t.exports.__esModule=!0),e(i)}t.exports=e,t.exports.default=t.exports,t.exports.__esModule=!0},,,function(t,f,e){"use strict";e.d(f,"b",function(){return i}),e.d(f,"c",function(){return r}),e.d(f,"a",function(){return c});var i=function(a){return function(d){var p=d.theme,h=d.size;return p.sizes[a][h]}},r=function(){var a=arguments.length>0&&arguments[0]!==void 0?arguments[0]:"&";return function(d){var p=d.theme,h=d.hasError;return`
      outline: none;
      box-shadow: 0;
      transition-property: border-color, box-shadow, fill;
      transition-duration: 0.2s;

      `.concat(a,`:focus-within {
        border: 1px solid `).concat(h?p.colors.danger600:p.colors.primary600,`;
        box-shadow: `).concat(h?p.colors.danger600:p.colors.primary600,` 0px 0px 0px 2px;
      }
    `)}},c=function(a){var d=a.theme;return`
  position: relative;
  outline: none;
  
  &:after {
    transition-property: all;
    transition-duration: 0.2s;
    border-radius: 8px;
    content: '';
    position: absolute;
    top: -4px;
    bottom: -4px;
    left: -4px;
    right: -4px;
    border: 2px solid transparent;
  }

  &:focus-visible {
    outline: none;
    &:after {
      border-radius: 8px;
      content: '';
      position: absolute;
      top: -5px;
      bottom: -5px;
      left: -5px;
      right: -5px;
      border: 2px solid `.concat(d.colors.primary600,`;
    }
  }
`)}},,,function(t,f,e){"use strict";var i=e(20);function r(){}function c(){}c.resetWarningCache=r,t.exports=function(){function a(h,y,w,u,l,x){if(x!==i){var O=new Error("Calling PropTypes validators directly is not supported by the `prop-types` package. Use PropTypes.checkPropTypes() to call them. Read more at http://fb.me/use-check-prop-types");throw O.name="Invariant Violation",O}}function d(){return a}a.isRequired=a;var p={array:a,bool:a,func:a,number:a,object:a,string:a,symbol:a,any:a,arrayOf:d,element:a,elementType:a,instanceOf:d,node:a,objectOf:d,oneOf:d,oneOfType:d,shape:d,exact:d,checkPropTypes:c,resetWarningCache:r};return p.PropTypes=p,p}},function(t,f,e){"use strict";t.exports="SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED"},function(t,f){t.exports=function(e,i){(i==null||i>e.length)&&(i=e.length);for(var r=0,c=new Array(i);r<i;r++)c[r]=e[r];return c},t.exports.default=t.exports,t.exports.__esModule=!0},function(t,f,e){var i=e(21);t.exports=function(r,c){if(r){if(typeof r=="string")return i(r,c);var a=Object.prototype.toString.call(r).slice(8,-1);return a==="Object"&&r.constructor&&(a=r.constructor.name),a==="Map"||a==="Set"?Array.from(r):a==="Arguments"||/^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(a)?i(r,c):void 0}},t.exports.default=t.exports,t.exports.__esModule=!0},,function(t,f){t.exports=function(e,i){if(e==null)return{};var r,c,a={},d=Object.keys(e);for(c=0;c<d.length;c++)r=d[c],i.indexOf(r)>=0||(a[r]=e[r]);return a},t.exports.default=t.exports,t.exports.__esModule=!0},function(t,f){t.exports=function(e){if(Array.isArray(e))return e},t.exports.default=t.exports,t.exports.__esModule=!0},function(t,f){t.exports=function(e,i){var r=e==null?null:typeof Symbol!="undefined"&&e[Symbol.iterator]||e["@@iterator"];if(r!=null){var c,a,d=[],p=!0,h=!1;try{for(r=r.call(e);!(p=(c=r.next()).done)&&(d.push(c.value),!i||d.length!==i);p=!0);}catch(y){h=!0,a=y}finally{try{p||r.return==null||r.return()}finally{if(h)throw a}}return d}},t.exports.default=t.exports,t.exports.__esModule=!0},function(t,f){t.exports=function(){throw new TypeError(`Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`)},t.exports.default=t.exports,t.exports.__esModule=!0},,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,function(t,f,e){"use strict";e.r(f),e.d(f,"TextButton",function(){return P});var i,r=e(5),c=e.n(r),a=e(4),d=e.n(a),p=e(3),h=e.n(p),y=e(1),w=e.n(y),u=e(0),l=e.n(u),x=e(2),O=e.n(x),T=e(6),S=e(8),n=e(9),o=e(16),s=["children","startIcon","endIcon","onClick","disabled"],C=O()(n.Flex)(i||(i=h()([`
  background: transparent;
  border: none;

  &[aria-disabled='true'] {
    pointer-events: none;
    svg path {
      fill: `,`;
    }
  }

  svg {
    display: flex;
    font-size: `,`rem;
  }

  svg path {
    fill: `,`;
  }

  `,`
`])),function(z){return z.theme.colors.neutral600},.625,function(z){return z.theme.colors.primary600},o.a),P=w.a.forwardRef(function(z,A){var H=z.children,M=z.startIcon,B=z.endIcon,E=z.onClick,k=z.disabled,W=d()(z,s),_=E&&!k?E:void 0;return w.a.createElement(C,c()({ref:A,"aria-disabled":k,onClick:_,as:"button",type:"button"},W),M&&w.a.createElement(T.Box,{as:"span",paddingRight:2,"aria-hidden":!0},M),w.a.createElement(S.Typography,{variant:"pi",textColor:k?"neutral600":"primary600"},H),B&&w.a.createElement(T.Box,{as:"span",paddingLeft:2,"aria-hidden":!0},B))});P.displayName="TextButton",P.defaultProps={disabled:!1,startIcon:void 0,endIcon:void 0,onClick:void 0},P.propTypes={children:l.a.node.isRequired,disabled:l.a.bool,endIcon:l.a.element,onClick:l.a.func,startIcon:l.a.element}}])})},56671:(v,j,g)=>{var m=g(35897),b=g(74741),t=b(m);v.exports=t},52920:(v,j,g)=>{var m=g(56671),b=g(81580);function t(f,e){var i=-1,r=b(f)?Array(f.length):[];return m(f,function(c,a,d){r[++i]=e(c,a,d)}),r}v.exports=t},50690:(v,j,g)=>{var m=g(26460),b=g(80166),t=g(88904),f=g(52920),e=g(36024),i=g(76535),r=g(1397),c=g(15362),a=g(77236);function d(p,h,y){h.length?h=m(h,function(l){return a(l)?function(x){return b(x,l.length===1?l[0]:l)}:l}):h=[c];var w=-1;h=m(h,i(t));var u=f(p,function(l,x,O){var T=m(h,function(S){return S(l)});return{criteria:T,index:++w,value:l}});return e(u,function(l,x){return r(l,x,y)})}v.exports=d},36024:v=>{function j(g,m){var b=g.length;for(g.sort(m);b--;)g[b]=g[b].value;return g}v.exports=j},40709:(v,j,g)=>{var m=g(22945);function b(t,f){if(t!==f){var e=t!==void 0,i=t===null,r=t===t,c=m(t),a=f!==void 0,d=f===null,p=f===f,h=m(f);if(!d&&!h&&!c&&t>f||c&&a&&p&&!d&&!h||i&&a&&p||!e&&p||!r)return 1;if(!i&&!c&&!h&&t<f||h&&e&&r&&!i&&!c||d&&e&&r||!a&&r||!p)return-1}return 0}v.exports=b},1397:(v,j,g)=>{var m=g(40709);function b(t,f,e){for(var i=-1,r=t.criteria,c=f.criteria,a=r.length,d=e.length;++i<a;){var p=m(r[i],c[i]);if(p){if(i>=d)return p;var h=e[i];return p*(h=="desc"?-1:1)}}return t.index-f.index}v.exports=b},74741:(v,j,g)=>{var m=g(81580);function b(t,f){return function(e,i){if(e==null)return e;if(!m(e))return t(e,i);for(var r=e.length,c=f?r:-1,a=Object(e);(f?c--:++c<r)&&i(a[c],c,a)!==!1;);return e}}v.exports=b},98978:(v,j,g)=>{var m=g(40314),b=g(50690),t=g(94569),f=g(63108),e=t(function(i,r){if(i==null)return[];var c=r.length;return c>1&&f(i,r[0],r[1])?r=[]:c>2&&f(r[0],r[1],r[2])&&(r=[r[0]]),b(i,m(r,1),[])});v.exports=e}}]);
