// @strapi/admin/build/cropper-css.0055cd53.chunk.js
"use strict";(self.webpackChunk_strapi_admin=self.webpackChunk_strapi_admin||[]).push([[7784],{56576:(c,r,n)=>{n.d(r,{Z:()=>u});var i=n(65064),o=n.n(i),e=n(38904),t=n.n(e),a=n(92128),d=n.n(a),_=new URL(n(8738),n.b),p=t()(o()),l=d()(_);p.push([c.id,`/*!
 * Cropper.js v1.5.11
 * https://fengyuanchen.github.io/cropperjs
 *
 * Copyright 2015-present Chen Fengyuan
 * Released under the MIT license
 *
 * Date: 2021-02-17T11:53:21.992Z
 */

.cropper-container {
  direction: ltr;
  font-size: 0;
  line-height: 0;
  position: relative;
  -ms-touch-action: none;
  touch-action: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.cropper-container img {
  display: block;
  height: 100%;
  image-orientation: 0deg;
  max-height: none !important;
  max-width: none !important;
  min-height: 0 !important;
  min-width: 0 !important;
  width: 100%;
}

.cropper-wrap-box,
.cropper-canvas,
.cropper-drag-box,
.cropper-crop-box,
.cropper-modal {
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
}

.cropper-wrap-box,
.cropper-canvas {
  overflow: hidden;
}

.cropper-drag-box {
  background-color: #fff;
  opacity: 0;
}

.cropper-modal {
  background-color: #000;
  opacity: 0.5;
}

.cropper-view-box {
  display: block;
  height: 100%;
  outline: 1px solid #39f;
  outline-color: rgba(51, 153, 255, 0.75);
  overflow: hidden;
  width: 100%;
}

.cropper-dashed {
  border: 0 dashed #eee;
  display: block;
  opacity: 0.5;
  position: absolute;
}

.cropper-dashed.dashed-h {
  border-bottom-width: 1px;
  border-top-width: 1px;
  height: calc(100% / 3);
  left: 0;
  top: calc(100% / 3);
  width: 100%;
}

.cropper-dashed.dashed-v {
  border-left-width: 1px;
  border-right-width: 1px;
  height: 100%;
  left: calc(100% / 3);
  top: 0;
  width: calc(100% / 3);
}

.cropper-center {
  display: block;
  height: 0;
  left: 50%;
  opacity: 0.75;
  position: absolute;
  top: 50%;
  width: 0;
}

.cropper-center::before,
.cropper-center::after {
  background-color: #eee;
  content: ' ';
  display: block;
  position: absolute;
}

.cropper-center::before {
  height: 1px;
  left: -3px;
  top: 0;
  width: 7px;
}

.cropper-center::after {
  height: 7px;
  left: 0;
  top: -3px;
  width: 1px;
}

.cropper-face,
.cropper-line,
.cropper-point {
  display: block;
  height: 100%;
  opacity: 0.1;
  position: absolute;
  width: 100%;
}

.cropper-face {
  background-color: #fff;
  left: 0;
  top: 0;
}

.cropper-line {
  background-color: #39f;
}

.cropper-line.line-e {
  cursor: ew-resize;
  right: -3px;
  top: 0;
  width: 5px;
}

.cropper-line.line-n {
  cursor: ns-resize;
  height: 5px;
  left: 0;
  top: -3px;
}

.cropper-line.line-w {
  cursor: ew-resize;
  left: -3px;
  top: 0;
  width: 5px;
}

.cropper-line.line-s {
  bottom: -3px;
  cursor: ns-resize;
  height: 5px;
  left: 0;
}

.cropper-point {
  background-color: #39f;
  height: 5px;
  opacity: 0.75;
  width: 5px;
}

.cropper-point.point-e {
  cursor: ew-resize;
  margin-top: -3px;
  right: -3px;
  top: 50%;
}

.cropper-point.point-n {
  cursor: ns-resize;
  left: 50%;
  margin-left: -3px;
  top: -3px;
}

.cropper-point.point-w {
  cursor: ew-resize;
  left: -3px;
  margin-top: -3px;
  top: 50%;
}

.cropper-point.point-s {
  bottom: -3px;
  cursor: s-resize;
  left: 50%;
  margin-left: -3px;
}

.cropper-point.point-ne {
  cursor: nesw-resize;
  right: -3px;
  top: -3px;
}

.cropper-point.point-nw {
  cursor: nwse-resize;
  left: -3px;
  top: -3px;
}

.cropper-point.point-sw {
  bottom: -3px;
  cursor: nesw-resize;
  left: -3px;
}

.cropper-point.point-se {
  bottom: -3px;
  cursor: nwse-resize;
  height: 20px;
  opacity: 1;
  right: -3px;
  width: 20px;
}

@media (min-width: 768px) {
  .cropper-point.point-se {
    height: 15px;
    width: 15px;
  }
}

@media (min-width: 992px) {
  .cropper-point.point-se {
    height: 10px;
    width: 10px;
  }
}

@media (min-width: 1200px) {
  .cropper-point.point-se {
    height: 5px;
    opacity: 0.75;
    width: 5px;
  }
}

.cropper-point.point-se::before {
  background-color: #39f;
  bottom: -50%;
  content: ' ';
  display: block;
  height: 200%;
  opacity: 0;
  position: absolute;
  right: -50%;
  width: 200%;
}

.cropper-invisible {
  opacity: 0;
}

.cropper-bg {
  background-image: url(`+l+`);
}

.cropper-hide {
  display: block;
  height: 0;
  position: absolute;
  width: 0;
}

.cropper-hidden {
  display: none !important;
}

.cropper-move {
  cursor: move;
}

.cropper-crop {
  cursor: crosshair;
}

.cropper-disabled .cropper-drag-box,
.cropper-disabled .cropper-face,
.cropper-disabled .cropper-line,
.cropper-disabled .cropper-point {
  cursor: not-allowed;
}
`,""]);const u=p},38904:c=>{c.exports=function(r){var n=[];return n.toString=function(){return this.map(function(o){var e="",t=typeof o[5]!="undefined";return o[4]&&(e+="@supports (".concat(o[4],") {")),o[2]&&(e+="@media ".concat(o[2]," {")),t&&(e+="@layer".concat(o[5].length>0?" ".concat(o[5]):""," {")),e+=r(o),t&&(e+="}"),o[2]&&(e+="}"),o[4]&&(e+="}"),e}).join("")},n.i=function(o,e,t,a,d){typeof o=="string"&&(o=[[null,o,void 0]]);var _={};if(t)for(var p=0;p<this.length;p++){var l=this[p][0];l!=null&&(_[l]=!0)}for(var u=0;u<o.length;u++){var s=[].concat(o[u]);t&&_[s[0]]||(typeof d!="undefined"&&(typeof s[5]=="undefined"||(s[1]="@layer".concat(s[5].length>0?" ".concat(s[5]):""," {").concat(s[1],"}")),s[5]=d),e&&(s[2]&&(s[1]="@media ".concat(s[2]," {").concat(s[1],"}")),s[2]=e),a&&(s[4]?(s[1]="@supports (".concat(s[4],") {").concat(s[1],"}"),s[4]=a):s[4]="".concat(a)),n.push(s))}},n}},92128:c=>{c.exports=function(r,n){return n||(n={}),r&&(r=String(r.__esModule?r.default:r),/^['"].*['"]$/.test(r)&&(r=r.slice(1,-1)),n.hash&&(r+=n.hash),/["'() \t\n]|(%20)/.test(r)||n.needQuotes?'"'.concat(r.replace(/"/g,'\\"').replace(/\n/g,"\\n"),'"'):r)}},65064:c=>{c.exports=function(r){return r[1]}},94300:(c,r,n)=>{n.r(r),n.d(r,{default:()=>m});var i=n(93379),o=n.n(i),e=n(7795),t=n.n(e),a=n(90569),d=n.n(a),_=n(3565),p=n.n(_),l=n(19216),u=n.n(l),s=n(44589),h=n.n(s),g=n(56576),f={};f.styleTagTransform=h(),f.setAttributes=p(),f.insert=d().bind(null,"head"),f.domAPI=t(),f.insertStyleElement=u();var A=o()(g.Z,f);const m=g.Z&&g.Z.locals?g.Z.locals:void 0},93379:c=>{var r=[];function n(e){for(var t=-1,a=0;a<r.length;a++)if(r[a].identifier===e){t=a;break}return t}function i(e,t){for(var a={},d=[],_=0;_<e.length;_++){var p=e[_],l=t.base?p[0]+t.base:p[0],u=a[l]||0,s="".concat(l," ").concat(u);a[l]=u+1;var h=n(s),g={css:p[1],media:p[2],sourceMap:p[3],supports:p[4],layer:p[5]};if(h!==-1)r[h].references++,r[h].updater(g);else{var f=o(g,t);t.byIndex=_,r.splice(_,0,{identifier:s,updater:f,references:1})}d.push(s)}return d}function o(e,t){var a=t.domAPI(t);a.update(e);var d=function(p){if(p){if(p.css===e.css&&p.media===e.media&&p.sourceMap===e.sourceMap&&p.supports===e.supports&&p.layer===e.layer)return;a.update(e=p)}else a.remove()};return d}c.exports=function(e,t){t=t||{},e=e||[];var a=i(e,t);return function(_){_=_||[];for(var p=0;p<a.length;p++){var l=a[p],u=n(l);r[u].references--}for(var s=i(_,t),h=0;h<a.length;h++){var g=a[h],f=n(g);r[f].references===0&&(r[f].updater(),r.splice(f,1))}a=s}}},90569:c=>{var r={};function n(o){if(typeof r[o]=="undefined"){var e=document.querySelector(o);if(window.HTMLIFrameElement&&e instanceof window.HTMLIFrameElement)try{e=e.contentDocument.head}catch(t){e=null}r[o]=e}return r[o]}function i(o,e){var t=n(o);if(!t)throw new Error("Couldn't find a style target. This probably means that the value for the 'insert' parameter is invalid.");t.appendChild(e)}c.exports=i},19216:c=>{function r(n){var i=document.createElement("style");return n.setAttributes(i,n.attributes),n.insert(i,n.options),i}c.exports=r},3565:(c,r,n)=>{function i(o){var e=n.nc;e&&o.setAttribute("nonce",e)}c.exports=i},7795:c=>{function r(o,e,t){var a="";t.supports&&(a+="@supports (".concat(t.supports,") {")),t.media&&(a+="@media ".concat(t.media," {"));var d=typeof t.layer!="undefined";d&&(a+="@layer".concat(t.layer.length>0?" ".concat(t.layer):""," {")),a+=t.css,d&&(a+="}"),t.media&&(a+="}"),t.supports&&(a+="}");var _=t.sourceMap;_&&typeof btoa!="undefined"&&(a+=`
/*# sourceMappingURL=data:application/json;base64,`.concat(btoa(unescape(encodeURIComponent(JSON.stringify(_))))," */")),e.styleTagTransform(a,o,e.options)}function n(o){if(o.parentNode===null)return!1;o.parentNode.removeChild(o)}function i(o){var e=o.insertStyleElement(o);return{update:function(a){r(e,o,a)},remove:function(){n(e)}}}c.exports=i},44589:c=>{function r(n,i){if(i.styleSheet)i.styleSheet.cssText=n;else{for(;i.firstChild;)i.removeChild(i.firstChild);i.appendChild(document.createTextNode(n))}}c.exports=r},8738:c=>{c.exports="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQAQMAAAAlPW0iAAAAA3NCSVQICAjb4U/gAAAABlBMVEXMzMz////TjRV2AAAACXBIWXMAAArrAAAK6wGCiw1aAAAAHHRFWHRTb2Z0d2FyZQBBZG9iZSBGaXJld29ya3MgQ1M26LyyjAAAABFJREFUCJlj+M/AgBVhF/0PAH6/D/HkDxOGAAAAAElFTkSuQmCC"}}]);
