// @strapi/admin/build/highlight.js.9d8ef460.chunk.js
"use strict";(self.webpackChunk_strapi_admin=self.webpackChunk_strapi_admin||[]).push([[7347],{27702:(d,s,n)=>{n.d(s,{Z:()=>u});var o=n(65064),a=n.n(o),e=n(38904),t=n.n(e),r=t()(a());r.push([d.id,`/*

Orginal Style from ethanschoonover.com/solarized (c) Jeremy Hull <sourdrums@gmail.com>

*/

.hljs {
  display: block;
  overflow-x: auto;
  padding: 0.5em;
  background: #002b36;
  color: #839496;
}

.hljs-comment,
.hljs-quote {
  color: #586e75;
}

/* Solarized Green */
.hljs-keyword,
.hljs-selector-tag,
.hljs-addition {
  color: #859900;
}

/* Solarized Cyan */
.hljs-number,
.hljs-string,
.hljs-meta .hljs-meta-string,
.hljs-literal,
.hljs-doctag,
.hljs-regexp {
  color: #2aa198;
}

/* Solarized Blue */
.hljs-title,
.hljs-section,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class {
  color: #268bd2;
}

/* Solarized Yellow */
.hljs-attribute,
.hljs-attr,
.hljs-variable,
.hljs-template-variable,
.hljs-class .hljs-title,
.hljs-type {
  color: #b58900;
}

/* Solarized Orange */
.hljs-symbol,
.hljs-bullet,
.hljs-subst,
.hljs-meta,
.hljs-meta .hljs-keyword,
.hljs-selector-attr,
.hljs-selector-pseudo,
.hljs-link {
  color: #cb4b16;
}

/* Solarized Red */
.hljs-built_in,
.hljs-deletion {
  color: #dc322f;
}

.hljs-formula {
  background: #073642;
}

.hljs-emphasis {
  font-style: italic;
}

.hljs-strong {
  font-weight: bold;
}
`,""]);const u=r},38904:d=>{d.exports=function(s){var n=[];return n.toString=function(){return this.map(function(a){var e="",t=typeof a[5]!="undefined";return a[4]&&(e+="@supports (".concat(a[4],") {")),a[2]&&(e+="@media ".concat(a[2]," {")),t&&(e+="@layer".concat(a[5].length>0?" ".concat(a[5]):""," {")),e+=s(a),t&&(e+="}"),a[2]&&(e+="}"),a[4]&&(e+="}"),e}).join("")},n.i=function(a,e,t,r,u){typeof a=="string"&&(a=[[null,a,void 0]]);var c={};if(t)for(var l=0;l<this.length;l++){var f=this[l][0];f!=null&&(c[f]=!0)}for(var h=0;h<a.length;h++){var _=[].concat(a[h]);t&&c[_[0]]||(typeof u!="undefined"&&(typeof _[5]=="undefined"||(_[1]="@layer".concat(_[5].length>0?" ".concat(_[5]):""," {").concat(_[1],"}")),_[5]=u),e&&(_[2]&&(_[1]="@media ".concat(_[2]," {").concat(_[1],"}")),_[2]=e),r&&(_[4]?(_[1]="@supports (".concat(_[4],") {").concat(_[1],"}"),_[4]=r):_[4]="".concat(r)),n.push(_))}},n}},65064:d=>{d.exports=function(s){return s[1]}},98362:(d,s,n)=>{n.r(s),n.d(s,{default:()=>v});var o=n(93379),a=n.n(o),e=n(7795),t=n.n(e),r=n(90569),u=n.n(r),c=n(3565),l=n.n(c),f=n(19216),h=n.n(f),_=n(44589),m=n.n(_),p=n(27702),i={};i.styleTagTransform=m(),i.setAttributes=l(),i.insert=u().bind(null,"head"),i.domAPI=t(),i.insertStyleElement=h();var E=a()(p.Z,i);const v=p.Z&&p.Z.locals?p.Z.locals:void 0},93379:d=>{var s=[];function n(e){for(var t=-1,r=0;r<s.length;r++)if(s[r].identifier===e){t=r;break}return t}function o(e,t){for(var r={},u=[],c=0;c<e.length;c++){var l=e[c],f=t.base?l[0]+t.base:l[0],h=r[f]||0,_="".concat(f," ").concat(h);r[f]=h+1;var m=n(_),p={css:l[1],media:l[2],sourceMap:l[3],supports:l[4],layer:l[5]};if(m!==-1)s[m].references++,s[m].updater(p);else{var i=a(p,t);t.byIndex=c,s.splice(c,0,{identifier:_,updater:i,references:1})}u.push(_)}return u}function a(e,t){var r=t.domAPI(t);r.update(e);var u=function(l){if(l){if(l.css===e.css&&l.media===e.media&&l.sourceMap===e.sourceMap&&l.supports===e.supports&&l.layer===e.layer)return;r.update(e=l)}else r.remove()};return u}d.exports=function(e,t){t=t||{},e=e||[];var r=o(e,t);return function(c){c=c||[];for(var l=0;l<r.length;l++){var f=r[l],h=n(f);s[h].references--}for(var _=o(c,t),m=0;m<r.length;m++){var p=r[m],i=n(p);s[i].references===0&&(s[i].updater(),s.splice(i,1))}r=_}}},90569:d=>{var s={};function n(a){if(typeof s[a]=="undefined"){var e=document.querySelector(a);if(window.HTMLIFrameElement&&e instanceof window.HTMLIFrameElement)try{e=e.contentDocument.head}catch(t){e=null}s[a]=e}return s[a]}function o(a,e){var t=n(a);if(!t)throw new Error("Couldn't find a style target. This probably means that the value for the 'insert' parameter is invalid.");t.appendChild(e)}d.exports=o},19216:d=>{function s(n){var o=document.createElement("style");return n.setAttributes(o,n.attributes),n.insert(o,n.options),o}d.exports=s},3565:(d,s,n)=>{function o(a){var e=n.nc;e&&a.setAttribute("nonce",e)}d.exports=o},7795:d=>{function s(a,e,t){var r="";t.supports&&(r+="@supports (".concat(t.supports,") {")),t.media&&(r+="@media ".concat(t.media," {"));var u=typeof t.layer!="undefined";u&&(r+="@layer".concat(t.layer.length>0?" ".concat(t.layer):""," {")),r+=t.css,u&&(r+="}"),t.media&&(r+="}"),t.supports&&(r+="}");var c=t.sourceMap;c&&typeof btoa!="undefined"&&(r+=`
/*# sourceMappingURL=data:application/json;base64,`.concat(btoa(unescape(encodeURIComponent(JSON.stringify(c))))," */")),e.styleTagTransform(r,a,e.options)}function n(a){if(a.parentNode===null)return!1;a.parentNode.removeChild(a)}function o(a){var e=a.insertStyleElement(a);return{update:function(r){s(e,a,r)},remove:function(){n(e)}}}d.exports=o},44589:d=>{function s(n,o){if(o.styleSheet)o.styleSheet.cssText=n;else{for(;o.firstChild;)o.removeChild(o.firstChild);o.appendChild(document.createTextNode(n))}}d.exports=s}}]);
